roles/config-pip-packages/README.md

```diff
@@ -0,0 +1,57 @@
+config-pip-packages
+===================
+
+A small role wrapping the Ansible pip module
+
+Requirements
+------------
+
+A Python interpreter
+
+Role Variables
+--------------
+
+The variables used must be defined in the Ansible Inventory using the `pip` dictionary as explained below.
+
+| Variable | Description | Required | Defaults |
+|:---------|:------------|:---------|:---------|
+|pip.packages|List of packages to install|yes|[]|
+|pip.virtualenv|The virtualenv to install packages in|no||
+|pip.virtualenv_python|The python executable used for virtualenv|no|None|
+
+Example Inventory
+-----------------
+
+```yaml
+---
+pip:
+  packages:
+    - awscli==1.6.210
+    - Flask==1.1.1
+    - ansible-tower-cli==3.3.6
+  virtualenv: /tmp/venv
+  virtualenv_python: python3.7
```

Comment:
Generally a flat variable design is better for ansible because of the way that ansible variable overrides behave.

If the variables are structured as:
```
pip_packages:
  - awscli==1.6.210
  - Flask==1.1.1
  - ansible-tower-cli==3.3.6
pip_virtualenv: /tmp/venv
pip_virtualenv_python: python3.7
```

Then the role can have defaults with an empty list for `pip_packages` and reasonable defaults for `pip_virtualenv` and `pip_virtualenv_python` and the user can choose to override just `pip_packages`. The flat variable structure also makes it simple to add other functionality with default variable values without breaking things.

If there is just a top level `pip` value then users have no choice but to set every required key under `pip` and new configuration options cannot be introduced easily without breaking existing usage.

Reply:
Although I'm not disagreeing with you re: general Ansible use, we are typically aiming for configurations to be kept in an inventory and not have individual values overridden at runtime - i.e.: aligning with IaC approach rather than creating snowflake configurations from single runs. Yes, it's useful to be able to override individual entries for testing/debugging, but not for longer term deployments. Hence, the historical aspect of using dictionaries to better organize the data in a repo for the purpose of IaC. For this particular PR, I believe it is just aligning with the other roles in this same area. We can have a bigger discussion around this and consider changes longer term, but probably wouldn't want to hold up this PR for the purpose of changing direction compared to the other roles that already exists. 

Also, although a bit more complex, it's possible to "combine" values at runtime the purpose of overriding/adding content to a data structure.

Reply:
I don't believe IaC approach comes into the consideration with this design. With any role it is still best practice to have default values in the role `default/main.yml` and only set in inventory the values that are customized for a particular environment.

Reply:
So, yeah, let's find some time for a bigger discussion around this instead of having a one-off role that's not aligned with the others.

Reply:
@jkupferer & @oybed thank you both for the feedback.